numpy/tests/test_public_api.py

```diff
@@ -94,6 +94,19 @@ def test_import_lazy_import(name):
     assert name in dir(np)
 
 
+@pytest.mark.parametrize('name', ['testing', 'Tester'])
+def test_dir_testing(name):
+    """
+    Test "dir" is not returning duplicate attribute for testing/Tester
+    """
+    exe = (sys.executable, '-c', "import numpy; numpy." + name)
```

Comment:
What is this for?

Reply:
This is to verify that explicitly calling numpy.testing is not adding a second copy of 'testing' to the result of dir.

Reply:
But this is started in a separate process, so can have no effect on the rest of the test.

Reply:
At this point in the test we probably already did anything bad (e.g. testing is definitely imported), wouldn't it be enough to just test that `len(dir(np)) == len(set(dir(np)))`, with a comment of why?

Reply:
@eric-wieser @seberg I have modified the test to assert len(dir(np)) == len(set(dir(np)))

Reply:
Thanks, seems fair enough to me. If you are still enthusiastic: We try to keep the lines below 80 characters long (so the comment is a bit too long). Might as well rebase/squash the commits (but we can also do both of that).

Reply:
@seberg  Thanks for the review. I have modified the comments to be inline with the standards.